src/kalman/WorldFilter.cpp

```diff
@@ -0,0 +1,156 @@
+//
+// Created by kjhertenberg on 13-5-19.
+//
+
+#include "roboteam_proto/messages_robocup_ssl_detection.pb.h"
+#include <kalman/WorldFilter.h>
+
+namespace world {
+
+    WorldFilter::WorldFilter() {
+        blueBots.clear();
+        yellowBots.clear();
+        balls.clear();
+    }
+
+// if we get a new frame we update our observations
+    void WorldFilter::addFrame(const proto::SSL_DetectionFrame &msg) {
+        const double filterGrabDistance=0.5;
```

Comment:
What it does is this: if an observation is within 0.5m of the state of the filter, we add it to the existing filter. Otherwise, we create a new filter, assuming it's a new object.

This allows us to track multiple objects of the same ID (such as multiple balls or double ID'd robots) provided they are far enough apart.  but also creates more stable vision. There may be a better metric than simply having a max distance like this, but this works for now.

Reply:
Note this is new behavior since we can now dynamically create filters whereas before any observation for one ID was applied to one ID and one ID only.